feat(ui): enhance merchant monitoring report status component

tomer-shvadron opened the pull request:
- Implement MerchantMonitoringReportStatus to display report statuses
- Update Button variants and import adjustments for optimization
- Refactor relevant table and business report components to utilize new status component


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a new `MerchantMonitoringReportStatus` component for managing merchant report statuses with dropdown and modal interfaces.
  - Added new `MerchantMonitoringStatusButton` and `MerchantMonitoringStatusBadge` components for enhanced status display.
  - Implemented a new API endpoint for updating the status of business reports.
  - Added a new `updateReportStatus` function for handling report status updates.

- **Style**
  - Enhanced button and badge visual variants for better user experience.

- **Refactor & Improvements**
  - Updated error handling mechanisms for report status updates.
  - Streamlined logic for rendering report statuses in tables and components.
  - Improved type definitions and validation for report status updates.

- **Chores**
  - Updated localization files to include new success and error messages for status updates.
  - Updated various dependencies to their latest versions for improved features and fixes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat(ui): enhance merchant monitoring report status component

- Implement MerchantMonitoringReportStatus to display report statuses
- Update Button variants and import adjustments for optimization
- Refactor relevant table and business report components to utilize new status component

MatanYadaev commented on apps/backoffice-v2/src/pages/MerchantMonitoring/components/MerchantMonitoringReportStatus/MerchantMonitoringReportStatus.tsx:
```diff
@@ -0,0 +1,127 @@
+};
+
+const statusToData = {
+  [MERCHANT_REPORT_STATUSES_MAP['in-progress']]: reportInProgressData,
```
Add `failed` as well?

tomer-shvadron replied:
Check it and there's no need, it is treated as `in-progress`

MatanYadaev commented on apps/backoffice-v2/src/pages/MerchantMonitoringBusinessReport/MerchantMonitoringBusinessReport.page.tsx:
```diff
@@ -109,7 +107,11 @@ export const MerchantMonitoringBusinessReport: FunctionComponent = () => {
     !isFetchingBusinessReport &&
     businessReport?.status &&
-    businessReport.status !== MERCHANT_REPORT_STATUSES_MAP['completed']
+    ![
+      MERCHANT_REPORT_STATUSES_MAP['pending-review'],
+      MERCHANT_REPORT_STATUSES_MAP['under-review'],
+      MERCHANT_REPORT_STATUSES_MAP.completed,
+    ].includes(businessReport?.status)
```
I think this array of statuses should be in the `@ballerine/common`

tomer-shvadron committed:
feat(reporting): enhance business report status management

- Add status update functionality to business reports
- Implement user feedback for status updates in toast notifications
- Create new UI components for report status and associated actions

tomer-shvadron committed:
feat(report-status): enhance Merchant Monitoring Report Status component

- Add support for a custom wrapper in react-table column metadata
- Implement dialog for completing reports with additional details
- Refactor component structure for improved readability and maintainability

fix(deps): update @radix-ui packages to latest versions

- Upgrade various @radix-ui libraries for bug fixes and performance improvements

style(dialog): adjust dialog for better user experience

- Prevent default behavior to enhance interaction on dialog close

tomer-shvadron committed:
chore(release): bump versions and update dependencies across projects

- bump versions
- Update dependencies for @ballerine/common and other related packages to latest versions
- Ensure consistency across CHANGELOGs with patch updates noted

tomer-shvadron committed:
Revert "chore(release): bump versions and update dependencies across projects"

This reverts commit 45d8d523ffe11c449ed21b98649a8b16ec1b9d0e.

tomer-shvadron committed:
Merge branch 'dev' into bal-1748-2

tomer-shvadron committed:
chore: update package versions and dependencies

- Bump versions for multiple packages including backoffice-v2 and kyb-app
- Update dependencies to latest compatible versions across the project

tomer-shvadron committed:
feat(merchant-monitoring): introduce updateable report statuses

- Add UPDATEABLE_REPORT_STATUSES constant for better status handling
- Refactor MerchantMonitoringReportStatus component to use new statuses
- Update useMerchantMonitoringTableLogic and business report logic
